test/test_dirichlet_dist.cpp

```diff
@@ -28,24 +32,17 @@
 
 using boost::math::dirichlet_distribution;
 using boost::math::concepts::real_concept;
-using std::numeric_limits;
 using std::domain_error;
-
-#define BOOST_TEST_MAIN
-#define BOOST_MATH_CHECK_THROW
+using std::numeric_limits;
 
 template <class RandomAccessContainer>
 void test_spot(
     RandomAccessContainer &&alpha,                      // concentration parameters 'a'
     RandomAccessContainer &&x,                          // quantiles 'x'
     RandomAccessContainer &&mean,                       // mean
-    RandomAccessContainer &&mode,                       // mode
     RandomAccessContainer &&var,                        // variance
-    RandomAccessContainer &&skewness,                   // skewness
-    RandomAccessContainer &&kurtosis,                   // kurtosis
     typename RandomAccessContainer::value_type entropy, // entropy
     typename RandomAccessContainer::value_type pdf,     // pdf
-    typename RandomAccessContainer::value_type cdf,     // cdf
     typename RandomAccessContainer::value_type tol)     // Test tolerance.
 {
```

Comment:
I removed ```mode```, ```skewness```, ```kurtosis```, ```cdf``` because I don't have predefined values for them. I thought of adding tests first for those that we have predefined values.